flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -705,9 +715,11 @@ public final void cancel() throws Exception {
 
         // the "cancel task" call must come first, but the cancelables must be
         // closed no matter what
+        FlinkUserSecurityManager.monitorSystemExitForCurrentThread();
         try {
             cancelTask();
```

Comment:
Just for documentation purposes: `cancelTask` indirectly calls `SourceFunction.cancel()`